src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1145,7 +1238,14 @@ private sealed class LoaderHolder
 
             // Verify that the current schema-defining arguments are default.
             // Get settings just for core arguments, not everything.
-            string tmp = CmdParser.GetSettings(host, args, new ArgumentsCore());
+            string tmp = CmdParser.GetSettings(host, options, new Options()
+            {
+                // It's fine if the user sets the following three arguments that are not considered core arguments.
+                // Setting the defaults to the user provided values will avoid these in the output of the call CmdParser.GetSettings.
```

Comment:
Why we override them here? Can you override them in upstream function?

Reply:
`GetSettings` builds a string with the fields in the second parameter `options` that are different from the fields in the third parameter in this case `new Options () {...}`. Previously, only the core arguments were considered when taking the difference between the user provided arguments and the defaults. Since now there is no distinction between core and non core arguments, I achieve exactly the same behavior as before by setting the non core arguments in the third parameter to the user provided values. 

---
In reply to: [255808097](https://github.com/dotnet/machinelearning/pull/2417#discussion_r255808097) [](ancestors = 255808097)